integration-tests/integration/prisonerProfile/prisonerCoursesAndQualificationsDetails.spec.js

```diff
@@ -17,12 +17,65 @@ context('Prisoner courses and qualifications details page', () => {
       cy.visit(`/prisoner/${offenderNo}/courses-qualifications`)
       cy.get('h1').should('have.text', 'John Smith’s courses and qualifications')
       cy.get('[data-test="no-courses-qualifications"]').should('exist')
+      cy.get('[data-test="courses-qualifications-returnLink"]').should('exist')
       cy.get('[data-test="no-courses-qualifications"]').should(
         'have.text',
         'John Smith has no courses or qualifications'
       )
     })
   })
+  context('One piece of data available', () => {
+    const dummyEducation = {
+      content: [
+        {
+          prn: 'G6123VU',
+          establishmentName: 'HMP Moorland',
+          courseName: 'Ocean Science',
+          isAccredited: false,
+          learningStartDate: '2021-02-13',
+          learningPlannedEndDate: '2021-06-08',
+          learningActualEndDate: null,
+          outcome: null,
+          outcomeGrade: null,
+          withdrawalReasons: null,
+          prisonWithdrawalReason: null,
+          completionStatus:
+            'The learner is continuing or intending to continue the learning activities leading to the learning aim',
+          withdrawalReasonAgreed: false,
+        },
+      ],
+    }
+    before(() => {
+      cy.clearCookies()
+      cy.task('reset')
+      cy.task('stubLogin', { username: 'ITAG_USER', caseload: 'MDI' })
+      cy.login()
+      cy.task('stubOffenderBasicDetails', { bookingId: 1, firstName: 'John', lastName: 'Smith', agencyId: 'MDI' })
+      cy.task('stubLearnerEducation', dummyEducation)
+    })
+
+    beforeEach(() => {
+      Cypress.Cookies.preserveOnce('hmpps-session-dev')
+    })
+
+    it('should show the correct content and number of results', () => {
+      cy.visit(`/prisoner/${offenderNo}/courses-qualifications`)
+      cy.get('h1').should('have.text', 'John Smith’s courses and qualifications')
+      cy.get('[data-test="courses-quals-result-number"]').should('have.text', 'Showing 1 result')
+      cy.get('[data-test="courses-qualifications-returnLink"]').should('exist')
+      cy.get('tbody tr')
+        .find('td')
+        .then(($cells) => {
+          expect($cells.length).to.eq(6)
```

Comment:
Don't worry about changing it now. But we normally extract the data into an object. A random example pulled from another test: `
const toOffender = ($cell) => ({
  name: $cell[1]?.textContent,
  prisonNo: $cell[2]?.textContent,
  location: $cell[3]?.textContent,
  alerts: $cell[4]?.textContent,
  cellHistoryText: $cell[5]?.textContent,
  changeCellText: $cell[6]?.textContent,
})`

Reply:
Oh interesting - hadn't seen that! Thanks